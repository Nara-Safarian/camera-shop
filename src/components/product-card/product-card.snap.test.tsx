import { render } from '@testing-library/react';
import ProductCard from './product-card';
import { BrowserRouter } from 'react-router-dom';

const product = {
  'id': 1,
  'name': 'Ретрокамера Dus Auge lV',
  'vendorCode': 'DA4IU67AD5',
  'type': 'Коллекционная',
  'category': 'Видеокамера',
  'description': 'Немецкий концерн BRW разработал видеокамеру Das Auge IV в начале 80-х годов, однако она до сих пор пользуется популярностью среди коллекционеров и яростных почитателей старинной техники. Вы тоже можете прикоснуться к волшебству аналоговой съёмки, заказав этот чудо-аппарат. Кто знает, может с Das Auge IV начнётся ваш путь к наградам всех престижных кинофестивалей.',
  'previewImg': 'img/content/das-auge.jpg',
  'level': 'Любительский',
  'price': 73450,
  'previewImg2x': 'img/content/das-auge@2x.jpg',
  'previewImgWebp': 'img/content/das-auge.webp',
  'previewImgWebp2x': 'img/content/das-auge@2x.webp',
  'reviewCount': 149
};

test('ProductCard snapshot should be rendered correctly', () => {
  const {container} = render(
    <BrowserRouter>
      <ProductCard product={product} />
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});
