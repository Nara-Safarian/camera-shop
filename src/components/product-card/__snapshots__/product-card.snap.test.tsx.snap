// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard snapshot should be rendered correctly 1`] = `
<div>
  <div
    class="product-card "
  >
    <div
      class="product-card__img"
    >
      <picture>
        <source
          srcset="img/content/das-auge.webp, img/content/das-auge@2x.webp 2x"
          type="image/webp"
        />
        <img
          alt="Ретрокамера Dus Auge lV"
          height="240"
          src="img/content/das-auge.jpg"
          srcset="img/content/das-auge@2x.jpg 2x"
          width="280"
        />
      </picture>
    </div>
    <div
      class="product-card__info"
    >
      <div
        class="rate product-card__rate"
      >
        <svg
          aria-hidden="true"
          height="16"
          width="17"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
        <svg
          aria-hidden="true"
          height="16"
          width="17"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
        <svg
          aria-hidden="true"
          height="16"
          width="17"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
        <svg
          aria-hidden="true"
          height="16"
          width="17"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
        <svg
          aria-hidden="true"
          height="16"
          width="17"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
        <p
          class="visually-hidden"
        >
          Рейтинг: 
          0
        </p>
        <p
          class="rate__count"
        >
          <span
            class="visually-hidden"
          >
            Всего оценок:
          </span>
          149
        </p>
      </div>
      <p
        class="product-card__title"
      >
        Ретрокамера Dus Auge lV
      </p>
      <p
        class="product-card__price"
      >
        <span
          class="visually-hidden"
        >
          Цена:
        </span>
        73450
         ₽
      </p>
    </div>
    <div
      class="product-card__buttons"
    >
      <button
        class="btn btn--purple product-card__btn"
        type="button"
      >
        Купить
      </button>
      <a
        class="btn btn--transparent"
        href="/cameras/1"
      >
        Подробнее
      </a>
    </div>
  </div>
</div>
`;
